config/config.go

```diff
@@ -298,3 +345,142 @@ func (rc *ReporterConfig) NewReporter(
 func (rc *ReporterConfig) newTransport() (jaeger.Transport, error) {
 	return jaeger.NewUDPTransport(rc.LocalAgentHostPort, 0)
 }
+
+// NewTracer returns a new tracer based on the current configuration, using the given options
+func (c *Configuration) NewTracer(options ...Option) (opentracing.Tracer, io.Closer, error) {
+	if c.ServiceName == "" {
+		return nil, nil, errors.New("no service name provided")
+	}
+
+	return c.New(c.ServiceName, options...)
+}
+
+// FromEnv uses environment variables to set the tracer's Configuration
+func FromEnv() *Configuration {
```

Comment:
nit - this is making the file huge, I suggest moving to `config_env.go` (including constants) for a nice encapsulation